src/usdb_syncer/usdb_scraper.py

```diff
@@ -571,3 +613,50 @@ def post_song_rating(song_id: SongId, stars: int) -> None:
     )
     logger = song_logger(song_id)
     logger.debug(f"{stars}-star rating posted on USDB.")
+
+
+def submit_local_changes(song: UsdbSong) -> None:
+    """Submit local changes of a song to USDB."""
+
+    logger = song_logger(song.song_id)
+
+    if not (sync_meta := song.sync_meta) or not (txt := sync_meta.txt):
+        logger.info("Song does not exist locally, skipping upload.")
+        return
+
+    if not (txt_to_upload := _prepare_txt_for_upload(sync_meta, logger)):
+        logger.warning("Failed to prepare song file, skipping upload.")
+        return
+
+    payload = {
+        "coverinput": "",
+        "sampleinput": song.sample_url,
+        "txt": txt_to_upload,
+        "filename": txt.fname,
+    }
+
+    get_usdb_page(
+        "index.php",
+        RequestMethod.POST,
+        headers={"Content-Type": "application/x-www-form-urlencoded"},
+        params={"link": "editsongsupdate", "id": str(song.song_id)},
+        payload=payload,
+    )
+
+    logger.info("Local changes successfully submitted to USDB.")
+
+
+def _prepare_txt_for_upload(sync_meta: SyncMeta, logger: Logger) -> str | None:
+    """Reinserts metatags and ensures CRLF line endings."""
+
+    txt_path = sync_meta.txt_path()
+    if not txt_path:
+        return None
+    if not (txt := SongTxt.try_from_file(txt_path, logger)):
+        logger.error("Failed to parse local song txt, skipping upload.")
+        return None
+    if txt.notes.track_2 is not None:
+        txt.headers.title += " [DUET]"
+    txt.headers.video = str(sync_meta.meta_tags)  # reinsert meta tags
```

Comment:
Would be nice if users could improve meta tags, too.

Reply:
True, didn't think of that. However, I guess the main use case is to download a song, then correct/improve the local file (in USDX, YASS, or even a text editor), then submit the changed file (that then does not contain the meta tags any longer). Do you have an idea how this could be implemented?